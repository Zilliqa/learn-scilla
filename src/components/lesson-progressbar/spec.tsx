import React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LessonProgressbar from '.';

const navigate = (chapterNum, lessonNum) => {
  console.log(`navigate to: /chapter/${chapterNum}/lesson/${lessonNum}`);
};

describe('Lesson Progressbar tests', () => {
  const baseComponent = () => (
    <LessonProgressbar navigate={navigate} chapterNumber={1} lessonNumber={5} total={10} />
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent()).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the component', () => {
      const wrapper = shallow(baseComponent());
      const assertion = wrapper.find('[data-test-id="lesson-progressbar"]').length;
      expect(assertion).toBe(1);
    });
  });
});
