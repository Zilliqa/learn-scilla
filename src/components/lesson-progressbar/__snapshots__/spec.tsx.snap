// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson Progressbar tests basic tests matches the snapshot 1`] = `
<div
  className="lesson-progress-bar-container progress"
>
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-primary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-primary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-primary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-primary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-primary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-secondary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-secondary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-secondary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-secondary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuenow={10}
    className="progress-bar lesson-progress-bar cursor-pointer bg-secondary"
    onClick={[Function]}
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;
